baybe/objectives/base.py

```diff
@@ -29,15 +29,29 @@ def targets(self) -> tuple[Target, ...]:
         """The targets included in the objective."""
 
     @abstractmethod
-    def transform(self, data: pd.DataFrame) -> pd.DataFrame:
+    def transform(
+        self,
+        df: pd.DataFrame,
+        /,
+        *,
+        allow_missing: bool = False,
+        allow_extra: bool = False,
+    ) -> pd.DataFrame:
         """Transform target values from experimental to computational representation.
 
         Args:
-            data: The data to be transformed. Must contain columns for all targets
-                but can contain additional columns.
+            df: The dataframe to be transformed. The allowed columns of the dataframe
+                are dictated by the ``allow_missing`` and ``allow_extra`` flags.
+            allow_missing: If ``False``, each target of the objective must have
+                (exactly) one corresponding column in the given dataframe. If ``True``,
+                the dataframe may contain only a subset of target columns.
+            allow_extra: If ``False``, every column present in the dataframe must
```

Comment:
Isn't it `each` instead of `every`?

Reply:
I'm not a 100% sure but I think in this situation they are interchangeable (can change to "each" nonetheless if you prefer). But would be glad to learn something new here. @Scienfitz, do you know for certain?

Reply:
nope dounds equiv to me

Reply:
ok, thx. Have changed it nevertheless to please @AVHopp 🐶

Reply:
See here some details on each vs. every: https://dictionary.cambridge.org/grammar/british-grammar/each-or-every I think in 99% of cases it is actually the same resp. the difference is so subtle that it does not matter.